src/chart/generateCategoricalChart.tsx

```diff
@@ -2152,7 +2153,7 @@ export const generateCategoricalChart = ({
         return typeof tooltipAxis.dataKey === 'function' ? tooltipAxis.dataKey(entry.payload) : null;
       }
 
-      if (hasActive) {
+      if (hasActiveWithAxisDefined) {
```

Comment:
The calculation is:

```typescript
      const hasActive = !hide && isTooltipActive && tooltipItem && (activeDot || activeBar || activeShape);
      
      const hasActiveWithAxisDefined = hasActive && activeTooltipIndex >= 0;
      
if (hasActiveWithAxisDefined) 
```
Admittedly, this feels off because it's based on `this.state.activeTooltipIndex` being defined which implicitly gets set by the mouse events via `<XAxis/>` and `<YAxis />`. Though, I'm not sure if the problem is with my naming choice or with the way I'm thinking about the problem.

Reply:
I would find it easier to read / understand if it was 
```

const hasActive = !hide && isTooltipActive && tooltipItem && (activeDot || activeBar || activeShape);
      
const hasActiveWithAxisDefined = hasActive && activeTooltipIndex >= 0;
      
if (hasActive) {

  if(activeTooltipIndex >= 0){
     ...
  }
  else {
     ...
  }
  ...
}
```